src/main/java/com/board/config/boardconfiggui/HomeViewController.java

```diff
@@ -42,14 +64,20 @@ public void onConfigureClick(String xmlFolderPath) {
         contentArea.getChildren().setAll(fxml);
     }
 
+    public void savePathData(String repositoryFolderPath, String toolChainFolderPath, String outputLocationFolderPath) {
+        this.repositoryFolderPath = repositoryFolderPath;
+        this.outputLocationFolderPath = outputLocationFolderPath;
+        this.toolChainFolderPath = toolChainFolderPath;
+    }
+
     public void onOutputGenerateClick() {
         loadDataView();
     }
 
     private void loadDataView() {
         Parent fxml = null;
         FXMLLoader loader = new FXMLLoader(getClass().getResource("load-data-view.fxml"));
-        LoadDataController loadDataController = new LoadDataController(this, xmlFolderPath);
+        LoadDataController loadDataController = new LoadDataController(this, xmlFolderPath, repositoryFolderPath, toolChainFolderPath, outputLocationFolderPath);
```

Comment:
Instead of passing these many arguments, better to use an object that should store all these paths.